### DISCUSSION TITLE
Need to restore the page number and page size of pagination after manual browser refresh

### DISCUSSION THREAD

**Discussion by Waqqars987** [TanStack/table#2786] [Q&A] [answered]
Hi,

I have the following requirement:

Suppose a user switches to **page number 5** with **page size 10** of my data table and manually refreshes the browser. Now, I want the user to be back on the previous **page number 5** itself with **page size 10**. What would be the correct approach to achieve this? 

Any help would be much appreciated.

**Comment by thedaviddias** [comment, marked as answer]
Did you consider to save these options in localStorage? Would probably be the easiest way to handle this.

**Reply by Waqqars987** [reply]
So, the initialState (pageIndex and pageSize) provided to the useTable hook would read directly from the localStorage?
Can you provided an example?

**Reply by Waqqars987** [reply]
I have been able to fulfil my requirements using sessionStorage. Thanks a lot.

**Reply by Myzel394** [reply]
@Waqqars987 https://xkcd.com/979/

What did you do? How did you implement this? What did you see on Oct 21. 2020 @Waqqars987 - whaaaaaaaaaat?

**Comment by michael-land** [comment]
You can add states to the url if you build a page table

**Reply by Waqqars987** [reply]
Do you mean query params?
